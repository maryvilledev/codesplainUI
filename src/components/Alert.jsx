import React from 'react';
import { Dialog, FlatButton } from 'material-ui';

/*
<Alert /> renders as a <Dialog /> with an OK button. It displays the text
specified in the 'text' prop, and invokes the 'onClose' prop when it is closed.
Setting the 'isError' prop to true causes its background to be rendered as a
crimson red.
*/
const Alert = ({ text, onClose }) => (
    <Dialog 
      actions={
        <FlatButton
          label="OK"
          primary={true}
          onTouchTap={onClose}
        />
      }
      modal={false}
      open={true}
      onRequestClose={onClose}
    >
      {text}
    </Dialog>
);

export default Alert
