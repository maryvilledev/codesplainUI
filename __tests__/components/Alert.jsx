import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

import Alert from '../../src/components/Alert';

describe('<Alert />', () => {
  it('matches the stored snapshot', () => {
    const renderer = ReactTestUtils.createRenderer();
    const tree = renderer.render(
      <Alert />
    );
    expect(tree).toMatchSnapshot();
  });
});
