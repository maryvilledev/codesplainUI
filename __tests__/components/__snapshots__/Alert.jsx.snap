exports[`<Alert /> matches the stored snapshot 1`] = `
<Dialog
  actions={
    <FlatButton
      disabled={false}
      label="OK"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      onTouchTap={undefined}
      primary={true}
      secondary={false} />
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={false}
  onRequestClose={undefined}
  open={true}
  repositionOnUpdate={true} />
`;
